location/models.py

```diff
@@ -159,14 +161,44 @@ def get_location_from_ids(self, qsr, loc_type):
             return [x for x in list(qsr) if x.type == loc_type]
         return list(qsr)
     
-    def is_allowed(self, user, locations_id):
+    def is_allowed(self, user, locations_id, strict=True):
+        if user.is_superuser or not settings.ROW_SECURITY:
+            return True 
         if hasattr(user, '_u'):
             user = user._u
-        cache_name = f'user_locations_{user.id}'
+        cache_name = f'user_locations_{"s" if strict else ""}_{user.id}'
         allowed = cache.get(cache_name)
         if not allowed:
-            allowed = list(self.allowed(user.id, qs=None))
-            allowed = [l.id for l in allowed]
+            is_list_id = False
+            # for CA
+            if user.is_claim_admin and user.health_facility:
+                allowed = set(self.children(
+                    location_id=user.health_facility.location_id
+                ))
+                if strict is False:
+                    allowed.update(self.parents(
+                        location_id=user.health_facility.location_id
+                    ))
+            elif user.is_officer:
+                is_list_id = True
+                allowed = list(OfficerVillage.objects.filter(
+                    officer=core_models.Officer.objects.filter(
+                        code=user.login_name,
+                        *filter_validity()
+                    ).first(),
+                    *filter_validity()
+                ).values_list('location_id', flat=True))
+                
+                if strict is False:
+                    parent = set(l.id for l in self.parents(
+                        location_id=allowed,
+                    ))
+                    parent.update(allowed)
+                    allowed = list(parent)
```

Comment:
I think this implementation of strict deviates from the implementation in `allowed`. In `allowed` when `strict=True`, it only includes parent locations if all their child locations are assigned to the user. When `strict=False`, `allowed` includes parent locations even if only some of their child locations are assigned to the user. Here `strict` is used differently. I'd suggest keeping it consistent.